src/storage/src/hummock/store/version.rs

```diff
@@ -879,24 +1030,30 @@ impl HummockVersionReader {
                     continue;
                 }
                 if sstables.len() > 1 {
-                    non_overlapping_iters.push(ConcatIterator::new(
-                        sstables,
-                        self.sstable_store.clone(),
-                        sst_read_options.clone(),
-                    ));
+                    use crate::hummock::iterator::HummockIteratorDirection;
```

Comment:
I think the logic of creating iter from sst can be implemented in different factories instead of here in the common logic. We only need method `add_staging_sst` and `add_concat_sst`, which only adds sst to the `IteratorFactory`. 

In this way, we don't need to have special logic for backward iter, and we don't even need to add an associated type `SstableIteratorType` to the `IteratorFactory`. `sstable_store` and `sst_read_options` can be added to `IteratorFactory` when we initially create the factory.

Reply:
Good suggestion. I will move it to `IteratorFactory`.